roles/DCOS.agent_windows/tasks/main.yml

```diff
@@ -0,0 +1,121 @@
+- name: Windows | Defining DC/OS by using 'version' or, more specifically, 'image_commit'
+  set_fact:
+     dcos_version_specifier: "{{ dcos_version | default('1.13') }}"
+     ansible_password: '{{ hostvars[inventory_hostname].pass }}'
+
+
+- debug:
+    msg: "dcos_version_specifier = {{ dcos_version_specifier }} , pass = {{ ansible_password }}, bucket_name = {{ bucket_name }}"
+
+
+#- name: Windows | edit Powershell Start script with local IP address of Mesos Master
+#  replace:#
+#    path: /files/start.ps1
+#    regexp: 'local_ip_master'
+#    replace: '{{  hostvars[groups.masters[0]].inventory_hostname }}'
+#  delegate_to: localhost
+
+
+- name: Windows | create folder for temporaly files
+  file:
+    path: /files
+    state: directory
+  delegate_to: localhost
+
+
+- name: Windows | create sub folder with dcos_version
+  file:
+    path: /files/{{dcos_version_specifier}}
+    state: directory
+  delegate_to: localhost
+
+
+- name: Windows | get vc_redist.x64.exe file from s3 to localhost
+  get_url:
+    url: "http://{{ bucket_name }}/vc_redist.x64.exe"
+    dest: /files/vc_redist.x64.exe
+    mode: '0660'
+  delegate_to: localhost
+
+
+- name: Windows | get vmesos-binaries.zip from s3 to localhost
+  get_url:
+    url: "http://{{ bucket_name }}/{{ dcos_version_specifier }}/mesos-binaries.zip"
+    dest: "/files/{{ dcos_version_specifier }}/mesos-binaries.zip"
+    mode: '0440'
+  delegate_to: localhost
+
+
+- name: Windows | initialisation windows-node (with install.ps1)
+  win_copy: src=/files/{{ item }} dest=C:\
+  with_items:
+      - vc_redist.x64.exe
+      - "{{ dcos_version_specifier }}/mesos-binaries.zip"
+
+
+- name: Windows | install vc_redist
+  win_shell: |
+    cd C:\
+    .\vc_redist.x64.exe /passive /quite /install /norestart /log  .\vcRedist_install.log
```

Comment:
this will be run on every apply and therefore showing a change even if it was already installed. You need check if its installed and add a condition or if there is a file where you clearly could see the install was successfull you can use the `creates:` attribute

Reply:
agreed

Reply:
added args with `chdir:` and `creates:`